### PR TITLE
Feature/add update option

### PR DESCRIPTION
**Pull request by frolovsa**
closes issue #15 

new code first checks if the destination file exists and matches md5 sum with the source file. if the sums match file logged as present and no copy occurs.

this PR also corrects a typo in one of the earlier yaml test files when option 1 and option 2 for the source files was the same.

### CODE REVIEW

File: src/s3-copy.py [hunk 1]
```diff
@@ -62,26 +66,46 @@
...
+          break
+
+    # if source file exists but the destiantion doesnt, then copy
     try:
       destination_bucket.copy( source_dict, destination_key)
     except botocore.exceptions.ClientError as e:
       if e.response['Error']['Code'] == "404":
         # log that the file is missing and try next source
         logger_missing[si].write("s3://{}/{}\n".format(source_dict["Bucket"], source_dict["Key"]));
+        print(f" missing {si}")
```
Comment (HenryRWinterbottom) [inline src/s3-copy.py:91]: Where is `si` defined?

File: src/s3-copy.py [hunk 2]
```diff
@@ -62,26 +66,46 @@
...
       logger_success.write("s3://{}/{} -> s3://{}/{}\n".format(source_dict["Bucket"], source_dict["Key"], destination["bucket"], destination_key))
+      print(f" copy from {si}")
+      break
+  # increment time counter
   dr.increment(seconds = cycling_interval)
 print('done with the time loop')
 
 #----- close logging files
 print('closing loggers')
 logger_success.close()
```
Comment (HenryRWinterbottom) [inline src/s3-copy.py:107]: For my own information, what is the difference between `success` and `present`? What do both imply?

File: src/s3-copy.py (hunk 1 above)
Comment (frolovsa) [reply src/s3-copy.py:91]: its a loop counter for the source stream.

File: src/s3-copy.py (hunk 2 above)
Comment (frolovsa) [reply src/s3-copy.py:107]: great question. 
`success` means the file was successfully copied. `present` means the file was already there and matched the source file. 
see examples of my log files here:
`https://noaa-reanalyses-pds.s3.amazonaws.com/index.html#observations/reanalysis/conv/gefs-prepbufr/log/`

note that this file path will disappear in the next 30 minutes or so and reappear here:
`https://noaa-reanalyses-pds.s3.amazonaws.com/index.html#observations/reanalysis/conv/prepbufr/log/`

as we discussed, i am still trying to figure out what is the right way to keep track of the file movements. I am considering adding md5 sum to the end of the line for success and present messages.

**Comment by frolovsa** [comment]
@HenryWinterbottom-NOAA can i merge this?

**Review by HenryRWinterbottom — APPROVED** [review]
LGTM. Approved.

**Pull request merged**